File: DownloadFiles/TGNet/Chapter04/Chapter4DemoCode/C#/BooleanSwitchDemo/BooleanSwitchDemo/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Diagnostics;

namespace BooleanSwitchDemo
{
  public partial class Form1 : Form
  {
    public Form1()
    {
      InitializeComponent();
    }

    private void button1_Click(object sender, EventArgs e)
    {
      //make sure to add using System.Diagnostics; at the top of the code file
      BooleanSwitch BS1 = new BooleanSwitch("MySwitch", "Decription: Toggles Trace");
      TextWriterTraceListener objTWFile = new TextWriterTraceListener("c:\\SwitchDemo.txt");
      Trace.Listeners.Add(objTWFile);
      Trace.AutoFlush = true ;
      
      Trace.WriteLineIf(BS1.Enabled == true, "MyMessage", "MyCatagory");
      MessageBox.Show("Now look in SwitchDemo.txt when switch is on");
      objTWFile.Close();
      Trace.Listeners.Clear();
    }
  }
}

File: DownloadFiles/TGNet/Chapter04/Chapter4DemoCode/VB/BooleanSwitchDemo/BooleanSwitchDemo/Form1.vb
Public Class Form1

  Private Sub Button1_Click _
     (ByVal sender As System.Object, ByVal e As System.EventArgs) _
    Handles Button1.Click

    Dim BS1 As BooleanSwitch = New _
           BooleanSwitch("MySwitch", "Decription: Toggles Trace")
    Dim objTWFile As TextWriterTraceListener = New TextWriterTraceListener("c:\SwitchDemo.txt")
    Trace.Listeners.Add(objTWFile)
    Trace.AutoFlush = True
    Trace.WriteLineIf(BS1.Enabled = True, "MyMessage", "MyCatagory")
    MessageBox.Show("Now look in SwitchDemo.txt when switch is on")
    objTWFile.Close()
    Trace.Listeners.Clear()
  End Sub
End Class
